lib/puppet/provider/image_streamer_resource.rb

```diff
@@ -0,0 +1,57 @@
+################################################################################
+# (C) Copyright 2017 Hewlett Packard Enterprise Development LP
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+require_relative 'login'
+require_relative 'common'
+require_relative 'oneview_resource'
+require 'oneview-sdk'
+
+module Puppet
+  # Base provider for OneView resources
+  class ImageStreamerResource < Puppet::OneviewResource
+    desc 'Base provider for Image Streamer resources'
+
+    def initialize(value = {})
```

Comment:
Currently doing nothing here. Won't we want to initialize any variables on this? or are the ones initialized through OneviewResource enough? if so, this should be dropped.